### PR TITLE
Implement analysis migration for VDI

### PR DESCRIPTION
**Pull request by dmgaldi**
## Overview
Analysis migration plugin for UDIS -> VDI

## Testing
Pending VDI deployment to dev

### CODE REVIEW

File: Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/VDIMigrationPlugin.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+      missingFromVdiCount++;
+      return new TableRowInterfaces.RowResult<>(nextRow);
+    }
+
+    // Append UD prefix to VDI ID. The prefix is prepended in the view that maps stable VDI IDs to the unstable study
+    // ID, which is the currency of EDA.
+    final String vdiDatasetId = UD_DATASET_ID_PREFIX + vdiId;
+
+    // Create a copy with just the dataset ID updated to VDI counterpart.
+    AnalysisRow out = new AnalysisRow(nextRow.getAnalysisId(), vdiDatasetId, nextRow.getDescriptor(),
```
Comment (ryanrdoherty) [inline Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/VDIMigrationPlugin.java:44]: You don't have to do this here.  You could just make a datasetId setter and modify nextRow.  The incoming object is a throwaway to the "framework".  Might save time.

File: Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/VDIMigrationPlugin.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        nextRow.getNumFilters(), nextRow.getNumComputations(), nextRow.getNumVisualizations());
+
+    return new TableRowInterfaces.RowResult<>(out);
```
Comment (ryanrdoherty) [inline Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/VDIMigrationPlugin.java:47]: Want to set the result object's update flag to true here unless _writeToDb is false.

File: Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/VDIMigrationPlugin.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+    if (vdiId == null) {
+      LOG.warn("Unable to find legacy ID " + legacyUdId + " in the tinydb file.");
+      missingFromVdiCount++;
+      return new TableRowInterfaces.RowResult<>(nextRow);
+    }
+
+    // Append UD prefix to VDI ID. The prefix is prepended in the view that maps stable VDI IDs to the unstable study
+    // ID, which is the currency of EDA.
+    final String vdiDatasetId = UD_DATASET_ID_PREFIX + vdiId;
```
Comment (ryanrdoherty) [inline Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/VDIMigrationPlugin.java:41]: I thought we were planning to change the previx to VDIID_ or something once we moved to it?  Weren't there multiple hour-long conversations about VD vs VDID vs VDIID?

File: Model/src/test/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/VDIMigrationPluginTest.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    final List<String> args = List.of("--tinyDb=" + file.getPath());
+    migrationPlugin.configure(mockedModel, args);
+    TableRowInterfaces.RowResult<AnalysisRow> result = migrationPlugin.processRecord(
+        new AnalysisRow("x",
+            "EDAUD_1234",
+            new JSONObject(),
+            3,
+            4,
+            5));
+    Assert.assertEquals("EDAUD_123XyZ", result.getRow().getDatasetId());
```
Comment (ryanrdoherty) [inline Model/src/test/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/VDIMigrationPluginTest.java:39]: So maybe also check that update flag is true here.  And can add a case where it should not be true.

**Review by ryanrdoherty — CHANGES_REQUESTED** [review]

File: Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/VDIMigrationPlugin.java [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class VDIMigrationPlugin extends AbstractAnalysisUpdater {
+  private static final Logger LOG = Logger.getLogger(VDIMigrationPlugin.class);
+  public static final String UD_DATASET_ID_PREFIX = "EDAUD_";
+
+  private Map<String, String> legacyIdToVdiId;
+  private int missingFromVdiCount = 0;
```
[the remark this answers is not in the document]
Comment (dmgaldi) [reply Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/VDIMigrationPlugin.java:25]: Forgot the executor was multi-threaded, fixed

File: Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/VDIMigrationPlugin.java (hunk 3 above)
Comment (dmgaldi) [reply Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/VDIMigrationPlugin.java:41]: This is just for referring to EDA user datasets that are managed by VDI. I think keeping it as UD should be fine. We discussed in scrum this morning.

**Review by ryanrdoherty — APPROVED** [review]

**Pull request merged**